src/main/java/roomescape/reservation/domain/repository/MemberReservationRepository.java

```diff
@@ -0,0 +1,34 @@
+package roomescape.reservation.domain.repository;
+
+import java.time.LocalDate;
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import roomescape.member.domain.Member;
+import roomescape.reservation.domain.MemberReservation;
+import roomescape.reservation.domain.Reservation;
+import roomescape.reservation.domain.ReservationTime;
+import roomescape.reservation.domain.Theme;
+
+public interface MemberReservationRepository extends JpaRepository<MemberReservation, Long> {
+
+    @Query("""
+            SELECT mr
+            FROM MemberReservation mr
+            JOIN mr.reservation r
+            JOIN mr.member m
+            JOIN r.time t
+            JOIN r.theme th
+            WHERE (:memberId IS NULL OR m.id = :memberId) 
+                AND (:themeId IS NULL OR th.id = :themeId) 
+                AND :startDate <= r.date 
+                AND r.date <= :endDate
+            """)
```

Comment:
member_reservation 테이블로 인해
jpa 장점인 쿼리를 작성하지 않아도된다는 점을 그렇게 많이 못 누리고 있는것 같아요

Reply:
```suggestion
    @Query("""
        SELECT mr FROM MemberReservation mr
        JOIN mr.reservation r
        WHERE mr.member.id = :memberId
        AND r.date between :startDate and :endDate
        AND r.theme.id = :themeId
        """)
```

쿼리가 어떤걸 조회하는지 잘 이해가 안되긴합니다
위 방법으로 조회하는것과 어떤 차이가 있나요?

Reply:
해당 부분은 예약을 조회할 때 사용하는 부분인데 사용자가 특정 정보를 입력하지 않는 경우를 생각했었어요.

예를 들어서 특정 예약자에 대해서만 조회를 하거나, 아니면 특정 테마에 특정 시간 이후로의 예약만을 조회하는 경우와 같이요!

이런 경우를 위해서 NULL 인 경우도 조회가 되도록 위와 같이 쿼리를 작성했었습니다!😁

맨 처음에는 동적 쿼리를 작성하기 위해서 매우 복잡하게 쿼리를 처리해주었는데 위와 같이 해주면 훨씬 간단하게 처리되더라고요😊

Reply:
동적 쿼리를 위한 방법으로는  JpaSpecificationExecutor 를 사용해보는것도 방법일것 같네요
time은 쓰이지 않는데 어떤 용도로 join 하는건가요?